checklist/checkList.js

```diff
@@ -0,0 +1,81 @@
+app.directive('testDirec', function () {
+
+
+    function linkingFunction(scope, element, attrs) {
+    }
+
+    function controller($scope) {
+        $scope.showAlreadyAdded = false;
+        $scope.itemsList=[
+        {
+            item:"maria1",
+            submitted:false,
+            checked:false
+        },
+        {
+            item:"maria2",
+            submitted:false,
+            checked:false
+        },
+        {
+            item:"maria3",
+            submitted:true,
+            checked:false
+        }
+        ];
+
+        //adds on the list to submit
+        $scope.addItem =function(){
+            $scope.itemsList.push({
+                    item: $scope.newItem,
+                    submitted:false,
+                    checked:false
+                });
+             $scope.newItem="";
+        }
+        //filters the displayed items
+        $scope.filterSubmitted=function(item){
+            if($scope.showAlreadyAdded)
+                return true;
+            return !item.submitted;
+
+        }
+        //when chekbox is clicked change flag
+        $scope.showAlreadyAddedF=function(){
+                $scope.showAlreadyAdded=!$scope.showAlreadyAdded;
+        }
+        //checks and uncheks an item
+        $scope.checkItem=function(item){
```

Comment:
This isn't necessary.  You can attached ng-model to the checkbox to produce identical functionality.  Same with the function above.